components/Feed/Caption.tsx

```diff
@@ -0,0 +1,45 @@
+"use client";
+import { useEffect, useState } from "react";
+
+export default function Caption({ caption }: { caption: string | null }) {
+  // track if the caption is truncated
+  const [isCaptionTruncated, setIsCaptionTruncated] = useState<boolean>(false);
+  // caption shown on the video slide
+  const [displayCaption, setDisplayCaption] = useState<string | null>(caption);
+  // track if the caption is expanded
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+  const truncationCharacterLimit = 80;
+
+  // truncate the caption if it's is greater than truncationCharacterLimit
+  useEffect(() => {
```

Comment:
Since the caption isn't dynamic I don't think we need the `useEffect`, can just do the calculations inline